### ISSUE TITLE
little issue when trying to create meeting on server mode

### ISSUE THREAD

**Issue by mihai-satmarean** [kosgrz/mattermost-plugin-skype4business#9]
Hi, this is a very nice integration, we would like to use,

After initial installation and configuration, we have this error:

Mattermost channel: 
System
2:58 PM(Only visible to you)
×
An error occurred when creating the meeting. Please try again later.

mattermost logs:

"caller":"go-plugin/stream.go:15","msg":"invalid character '<' looking for beginning of value","plugin_id":"skype4business","source":"plugin_stdout"}

It could be that the configuration is not done right,
Thanks a lot for the nice integration.

**Comment by kosgrz** [comment]
Hi @mihai-satmarean! Thank you very much for your kind words! It's very motivating!

Logs definitely should be more descriptive. It will be fixed in the next release soon.

I had the same issue when I set invalid values in "bot username" or "bot password" fields in the plugin settings. Maybe it's the same in your case? Could you log in to normal Skype for Business client using these credentials?

**Comment by mihai-satmarean** [comment]
Hello, 

indeed changing a bit the domain lead to a different error

`{"level":"info","ts":1559829208.5631034,"caller":"go-plugin/stream.go:15","msg":"Get https://lyncdiscover.REDACTEDDOMAIN.com: dial tcp: lookup lyncdiscover.REDACTEDDOMAIN.com on 10.24.128.10:53: no such host","plugin_id":"skype4business","source":"plugin_stdout"}`

any clue there?

Keep up the good work, people need better integrations.

**Comment by kosgrz** [comment]
It means that the domain you set in the settings is wrong. This request to "https://lyncdiscover.REDACTEDDOMAIN.com" is the first step in the process of creation of a meeting.

I think you passed a correct domain initially. The error "invalid character '<' looking for beginning of value" could occur only after a successful response from the https://lyncdiscover.<YOUR DOMAIN>. I think the problem is with values in fields "bot username" and "bot password". Could you check them?

Anyway, I will upload a new release with better logging probably tomorrow and I hope we will see what causes that issue.

**Comment by kosgrz** [comment]
Hi @mihai-satmarean ! There is a new [release](https://github.com/kosgrz/mattermost-plugin-skype4business/releases/tag/v0.1.1) with improved logging. Could you install it on your server and copy logs here? It would be a great way to check what causes your issue.

**Comment by mihai-satmarean** [comment]
Hi,
Thanks!
I am off this week, will try next week.


În dum., 9 iun. 2019 la 19:32, kosgrz <notifications@github.com> a scris:

**Comment by DmitryVladimirov** [comment]
Hi!
why you use https://lyncdiscover.REDACTEDDOMAIN.com?
lyncdiscover.contoso.com - External record for client AutoDiscover, also used by Mobility, Skype for Business Web App, and scheduler Web app, resolved by the reverse proxy server

Mb use lyncdiscoverinternal.domain.com? 
and one more
what is lyncweb.domain.com? 
I cant find it
 https://docs.microsoft.com/en-us/skypeforbusiness/plan-your-deployment/network-requirements/dns

**Comment by kosgrz** [comment]
Hi @DmitryVladimirov! Thanks for the feedback!

> why you use https://lyncdiscover.REDACTEDDOMAIN.com?
> lyncdiscover.contoso.com - External record for client AutoDiscover, also used by Mobility, Skype for Business Web App, and scheduler Web app, resolved by the reverse proxy server

Good catch! I based my knowledge on the UCWA documentation from [there](https://ucwa.skype.com/documentation/GettingStarted-RootURL), but indeed I missed a case when a Mattermost Server is running in the network as the Skype for Business server. In that case, I should use  "lyncdiscoverinternal" instead of "lyncdiscover". It will be fixed immediately!

> and one more
> what is lyncweb.domain.com?

This is my fault too! It should be based on a response from one of the addresses provided by the lyncdiscover or lyncdiscoverinternal. I will fix it rapidly!

Thanks again for your great feedback! Would you like to test a prelease of the plugin with the fixes above? It would be really helpful! I can upload the prerelease tomorrow :rocket: .

**Comment by DmitryVladimirov** [comment]
@kosgrz  Hi!
Sure!


@mihai-satmarean Like workaroud you can write in /etc/hosts of Mattermosts server
lyncdiscover.domain.com - S4B frontend server ip
lyncweb.domain.com - S4B frontend server ip
it works for me.

**Comment by kosgrz** [comment]
Hi @DmitryVladimirov! I've just uploaded the prerelase with fixes. You can download it [here](https://github.com/kosgrz/mattermost-plugin-skype4business/releases/tag/v0.1.2). It works correctly in my test lab. I hope it will work for you too!

**Comment by DmitryVladimirov** [comment]
Hi! @kosgrz  
Thnx! Everything works fine!

**Comment by kosgrz** [comment]
Hi @DmitryVladimirov !
Great to hear that it works for you! :tada: Thanks a lot for the test!

> But I have another issue with created meetings.
> All users connect as attendees and cannot presenting desktop, programm, etc.

Thank you for this information! It's very important to make this plugin better! I created another issue in this repo to track it: #10

**Comment by kosgrz** [comment]
Hi @mihai-satmarean ! Have you tried the latest prelease of the plugin?

**Comment by mihai-satmarean** [comment]
Hi,
Thank you for the constant effort,
yes, I tried it right now, and I got this
`{"level":"info","ts":1560930365.8947322,"caller":"go-plugin/stream.go:15","msg":"{\"level\":\"error\",\"msg\":\"Error fetching meetings resource url: Error during authentication: Bad response received. Status: 400 Bad Request. X-Ms-Diagnostics from response: 28020;source=\\\"redacted.com\\\";reason=\\\"No valid security token.\\\". Response body: {\\\"error\\\":\\\"invalid_grant\\\"}. \"}","plugin_id":"skype4business","source":"plugin_stdout"}
`

**Comment by kosgrz** [comment]
@mihai-satmarean thanks for the reply! According to the [documentation](https://docs.microsoft.com/en-us/skype-sdk/ucwa/authenticationinucwa):
> **invalid_grant**: The credentials provided could not be verified.

it seems that your credentials are wrong. Have you tried to log in to normal Skype for Business client using these credentials?

**Comment by mihai-satmarean** [comment]
Hello,

I indeed use my credentials and they work in skype for business, I had changed the user to be of the form domain\user and now I have this response:

`"level":"info","ts":1561022675.3728287,"caller":"go-plugin/stream.go:15","msg":"{\"level\":\"error\",\"msg\":\"Error fetching meetings resource url: Error creating a new application: Bad response received. Status: 500 Internal Server Error. X-Ms-Diagnostics from response: 28032;source=\\\"redacted.com\\\";reason=\\\"The web ticket is invalid.\\\". Response body: \\u003c!DOCTYPE html PUBLIC \\\"-//W3C//DTD XHTML 1.0 Strict//EN\\\" \\\"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd\\\"\\u003e\\r\\n\\u003chtml xmlns=\\\"http://www.w3.org/1999/xhtml\\\"\\u003e\\r\\n\\u003chead\\u003e\\r\\n\\u003cmeta http-equiv=\\\"Content-Type\\\" content=\\\"text/html; charset=iso-8859-1\\\"/\\u003e\\r\\n\\u003ctitle\\u003e500 - Internal server error.\\u003c/title\\u003e\\r\\n\\u003cstyle type=\\\"text/css\\\"\\u003e\\r\\n\\u003c!--\\r\\nbody{margin:0;font-size:.7em;font-family:Verdana, Arial, Helvetica, sans-serif;background:#EEEEEE;}\\r\\nfieldset{padding:0 15px 10px 15px;} \\r\\nh1{font-size:2.4em;margin:0;color:#FFF;}\\r\\nh2{font-size:1.7em;margin:0;color:#CC0000;} \\r\\nh3{font-size:1.2em;margin:10px 0 0 0;color:#000000;} \\r\\n#header{width:96%;margin:0 0 0 0;padding:6px 2% 6px 2%;font-family:\\\"trebuchet MS\\\", Verdana, sans-serif;color:#FFF;\\r\\nbackground-color:#555555;}\\r\\n#content{margin:0 0 0 2%;position:relative;}\\r\\n.content-container{background:#FFF;width:96%;margin-top:8px;padding:10px;position:relative;}\\r\\n--\\u003e\\r\\n\\u003c/style\\u003e\\r\\n\\u003c/head\\u003e\\r\\n\\u003cbody\\u003e\\r\\n\\u003cdiv id=\\\"header\\\"\\u003e\\u003ch1\\u003eServer Error\\u003c/h1\\u003e\\u003c/div\\u003e\\r\\n\\u003cdiv id=\\\"content\\\"\\u003e\\r\\n \\u003cdiv class=\\\"content-container\\\"\\u003e\\u003cfieldset\\u003e\\r\\n  \\u003ch2\\u003e500 - Internal server error.\\u003c/h2\\u003e\\r\\n  \\u003ch3\\u003eThere is a problem with the resource you are looking for, and it cannot be displayed.\\u003c/h3\\u003e\\r\\n \\u003c/fieldset\\u003e\\u003c/div\\u003e\\r\\n\\u003c/div\\u003e\\r\\n\\u003c/body\\u003e\\r\\n\\u003c/html\\u003e\\r\\n. \"}","plugin_id":"skype4business","source":"plugin_stdout"}`

**Comment by kosgrz** [comment]
Hi @mihai-satmarean !
Could you try to install and test the prerelase of the plugin? I think the above issue is fixed there. Here you can download the prerelease:
https://github.com/kosgrz/mattermost-plugin-skype4business/releases/tag/v0.1.2

Please let me know if it works. If not, I will try to figure it out :+1: .

**Comment by mihai-satmarean** [comment]
Hi,

I think I had already installed that one before sending the logs.
Actually I tried installing it again and the mattermost asked me to accept
an overwrite because existing id.
Maybe you need to update the version and the ID with each change, also the
tags do not match.

Thanks again for the continued effort.
This plugin could change the way we use mattermost.



În vin., 21 iun. 2019 la 03:21, kosgrz <notifications@github.com> a scris:

**Comment by kosgrz** [comment]
@mihai-satmarean thanks for the information about versions and fix. I will remember about that next time :+1:.

This issue is a little more difficult than I thought initially. I try to send a fix before the weekend.

**Comment by kosgrz** [comment]
@mihai-satmarean I made some fixes. I hope it will work now. There is the updated release:
https://github.com/kosgrz/mattermost-plugin-skype4business/releases/tag/v0.1.2

Let me know if it works.

**Comment by mihai-satmarean** [comment]
Hi, please add the compiled version too.

În dum., 30 iun. 2019 la 04:59, kosgrz <notifications@github.com> a scris:

**Comment by kosgrz** [comment]
@mihai-satmarean sorry, I've uploaded the compiled version.

**Comment by mihai-satmarean** [comment]
it could be that I am doing something wrong?

the result is this:
`{"level":"error","ts":1561966412.7121542,"caller":"log/log.go:296","msg":"forbidden content-type: \"text/html\"","source":"stdlog"}
{"level":"error","ts":1561966412.7128594,"caller":"log/log.go:296","msg":"forbidden content-type: \"text/html\"","source":"stdlog"}
{"level":"error","ts":1561966412.7136042,"caller":"log/log.go:296","msg":"forbidden content-type: \"text/html\"","source":"stdlog"}
{"level":"info","ts":1561966415.2030792,"caller":"go-plugin/stream.go:15","msg":"{\"level\":\"warn\",\"msg\":\"Resource from applications url is not the same as resource name from user url\"}","plugin_id":"skype4business","source":"plugin_stdout"}
{"level":"info","ts":1561966415.2370276,"caller":"go-plugin/stream.go:15","msg":"{\"level\":\"error\",\"msg\":\"Error fetching meetings resource url: Error creating a new application: Bad response received. Status: 500 Internal Server Error. X-Ms-Diagnostics from response: 28032;source=\\\"redacted.com\\\";reason=\\\"The web ticket is invalid.\\\". Response body: \\u003c!DOCTYPE html PUBLIC \\\"-//W3C//DTD XHTML 1.0 Strict//EN\\\" \\\"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd\\\"\\u003e\\r\\n\\u003chtml xmlns=\\\"http://www.w3.org/1999/xhtml\\\"\\u003e\\r\\n\\u003chead\\u003e\\r\\n\\u003cmeta http-equiv=\\\"Content-Type\\\" content=\\\"text/html; charset=iso-8859-1\\\"/\\u003e\\r\\n\\u003ctitle\\u003e500 - Internal server error.\\u003c/title\\u003e\\r\\n\\u003cstyle type=\\\"text/css\\\"\\u003e\\r\\n\\u003c!--\\r\\nbody{margin:0;font-size:.7em;font-family:Verdana, Arial, Helvetica, sans-serif;background:#EEEEEE;}\\r\\nfieldset{padding:0 15px 10px 15px;} \\r\\nh1{font-size:2.4em;margin:0;color:#FFF;}\\r\\nh2{font-size:1.7em;margin:0;color:#CC0000;} \\r\\nh3{font-size:1.2em;margin:10px 0 0 0;color:#000000;} \\r\\n#header{width:96%;margin:0 0 0 0;padding:6px 2% 6px 2%;font-family:\\\"trebuchet MS\\\", Verdana, sans-serif;color:#FFF;\\r\\nbackground-color:#555555;}\\r\\n#content{margin:0 0 0 2%;position:relative;}\\r\\n.content-container{background:#FFF;width:96%;margin-top:8px;padding:10px;position:relative;}\\r\\n--\\u003e\\r\\n\\u003c/style\\u003e\\r\\n\\u003c/head\\u003e\\r\\n\\u003cbody\\u003e\\r\\n\\u003cdiv id=\\\"header\\\"\\u003e\\u003ch1\\u003eServer Error\\u003c/h1\\u003e\\u003c/div\\u003e\\r\\n\\u003cdiv id=\\\"content\\\"\\u003e\\r\\n \\u003cdiv class=\\\"content-container\\\"\\u003e\\u003cfieldset\\u003e\\r\\n  \\u003ch2\\u003e500 - Internal server error.\\u003c/h2\\u003e\\r\\n  \\u003ch3\\u003eThere is a problem with the resource you are looking for, and it cannot be displayed.\\u003c/h3\\u003e\\r\\n \\u003c/fieldset\\u003e\\u003c/div\\u003e\\r\\n\\u003c/div\\u003e\\r\\n\\u003c/body\\u003e\\r\\n\\u003c/html\\u003e\\r\\n. \"}","plugin_id":"skype4business","source":"plugin_stdout"}
{"level":"info","ts":1561966426.2729075,"caller":"scheduler/worker.go:78","msg":"Worker: Job is complete","worker":"Plugins","job_id":"xdt469rhwbr6jc4tyogh9ps58y"}`

**Comment by kosgrz** [comment]
@mihai-satmarean thank you for the logs. I made some fixes and hope it will work this time. You can download it here:
https://github.com/kosgrz/mattermost-plugin-skype4business/releases/tag/v0.1.2

**Comment by mihai-satmarean** [comment]
thanks for your efforts, actually the message is the same, and the version is the same, always asks me to overwrite. maybe you pushed the old code again?

**Comment by kosgrz** [comment]
@mihai-satmarean I'm still fighting :D! I'm not quite sure what is the cause of the above issues, so I need to do some attempts. There is the latest release:
https://github.com/kosgrz/mattermost-plugin-skype4business/releases/tag/v0.1.2-1
Would you like to test it and put logs here?

**Comment by mihai-satmarean** [comment]
Congratulations! You fixed it! I was able to create a meeting and join it by clicking the  button.
